_posts/en/newsletters/2020-09-16-newsletter.md

```diff
@@ -0,0 +1,224 @@
+  many of these inventories were queued, the victim's node would run out
+  of memory and crash.
+
+    Several of the affected programs had code that limited the number of
+    inventories that should've been queued, but the researchers were
+    able to circumvent those protections.  As a bug that could only crash
+    nodes, it could not be used to directly steal from affected nodes.
+    However, it could be used to execute [eclipse attacks][topic eclipse
+    attacks] that could ultimately steal money---although there's no
+    indication the attack was ever used in public.
+
+    All users are encouraged to upgrade to the latest released version
+    of their preferred full node software or to review the list of
+    vulnerable and patched versions on the [website][invdos site].
+
+- **Stealing onchain fees from LN HTLCs:** Antoine Riard [posted][riard
+  post] to the Lightning-Dev mailing list a potential vulnerability in a
+  recent update to the LN specification (see [Newsletter #112][news112
+  bolts688]). When LN payments are made according to the updated
+  specification, each remote party includes a [flag][sighash] in the signature for
+  their half of the 2-of-2 multisig script.  The flag, called
+  `SIGHASH_SINGLE|SIGHASH_ANYONECANPAY`, allows the payment (called an
+  HTLC) to be placed into a transaction containing other inputs and
+  outputs.  This is designed to allow multiple HTLCs  to be bundled
+  together into a transaction that also contains an extra input that
+  pays any additional transaction fees and an optional extra output that
+  collects any money left over from paying those extra fees.  This makes
+  it possible to wait until those HTLCs need to be confirmed before
+  choosing what feerate to pay.
+
+    {:.center}
+    ![Spending HTLCs with a fee-bumping input and a change output](/img/posts/2020-09-htlc-fee-bumping.dot.png)
+
+    However, Riard notes that the specification's earlier mechanism of
+    committing to variable feerates for HTLCs is still available.  This
+    can allow an attacker to commit to a high fee for multiple HTLCs
+    and then construct a transaction that combines those HTLCs with just
+    an additional output that claims some of the funds expected to be used as fees.  Riard's email
+    describes how to use this basic technique in a more complex attack
+    to maximize the amount stolen.
+
+    {:.center}
+    ![Using a change output to steal fees from HTLCs that overpay fees](/img/posts/2020-09-htlc-fee-stealing.dot.png)
+
+    Several solutions were proposed, the simplest of which might be to
+    have HTLCs only pay a minimal relay fee---requiring the party
+    broadcasting the HTLCs add any additional necessary fees.  No LN
+    implementation defaults to using this recent specification update
+    yet, so only users who have played with experimental options related
+    to [anchor outputs][topic anchor outputs] should be affected.
```

Comment:
I find it a bit odd that anchor outputs are stated here at the end but not from the start in line 59 ("recent update to the LN specification").

Reply:
Yeah, @bitschmidty had the [same thought](https://github.com/bitcoinops/bitcoinops.github.io/pull/458#discussion_r488037514).  I still think it's a bit confusing to mention anchors early in this item because the problem itself has nothing to do with anchor outputs themselves but rather a change that was made in the same BOLT update as anchors.

Reply:
Oh, didn't read closely enough to realise that. Maybe clarify with something along the lines of your comment: "the vulnerability has nothing to do with anchor outputs themselves, but rather a change (link-to-BOLT-spec) that was made in the same BOLT update."